numpy/core/arrayprint.py

```diff
@@ -1453,10 +1457,14 @@ def dtype_short_repr(dtype):
         return "'%s'" % str(dtype)
 
     typename = dtype.name
+    if not dtype.isnative:
+        # deal with cases like dtype('<u2') that are identical to an
+        # established dtype (in this case uint16)
+        # except that they have a different endianness.
+        return repr(dtype.descr[0][1])
```

Comment:
Using `descr` feels like a mistake to me; `descr` is generally poorly behaved. I thin kthere should be another way to get this information?

Reply:
`_construction_repr(..., short=True)` from https://github.com/numpy/numpy/blob/main/numpy/core/_dtype.py looks like it might do the right thing.

Reply:
I'll look into that. `descr` was a very ad-hoc choice made because it looked like the right kind of thing.

Reply:
Yeah, the default in this branch looks like using `str()` or `repr()` one of them should be right (plausibly both, because complicated dtypes are dealt with earlier).

Reply:
Given that all tests pass when using `_construction_repr(dtype)` in that conditional branch (the current status as of my most recent commit), should I leave as is, or do you think it's better to switch back?

Reply:
That seems still very roundabout (maybe it is actually more direct, but using more internals), and `str` is already being used for structured dtypes?